utils/defaulter.go

```diff
@@ -1,39 +1,68 @@
 package utils
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 
 	"github.com/imdario/mergo"
+	"github.com/kong/go-kong/kong"
 )
 
 // Defaulter registers types and fills in struct fields with
 // default values.
 type Defaulter struct {
 	r map[string]interface{}
+
+	ctx    context.Context
+	client *kong.Client
+
+	service  *kong.Service
+	route    *kong.Route
+	upstream *kong.Upstream
+	target   *kong.Target
+}
+
+// NewDefaulter initializes a Defaulter with empty entities.
+func NewDefaulter() *Defaulter {
+	return &Defaulter{
+		service:  &kong.Service{},
+		route:    &kong.Route{},
+		upstream: &kong.Upstream{},
+		target:   &kong.Target{},
+	}
 }
 
 // GetKongDefaulter returns a defaulter which can set default values
 // for Kong entities.
-func GetKongDefaulter() (*Defaulter, error) {
-	var d Defaulter
-	err := d.Register(&serviceDefaults)
+func GetKongDefaulter(kongDefaults interface{}, isKonnect bool) (*Defaulter, error) {
```

Comment:
Take input DefaulterOpts with these two.

Rename isKonnect to disableDynamicDefaults

Reply:
done